src/nyaml/nxdl2nyaml.py

```diff
@@ -741,6 +750,13 @@ def handle_dim_with_all_dim_attr(depth, node, file_out, possible_dim_attrs):
                     # that it does not call in xmlparser function
                     dim_cmnt_nodes.append(child)
                     node.remove(child)
+            # special case for NXdata.nxdl.xml's field DATA where rank is a symbol
+            if (
+                len(dim_index_value) == 0
+                and len(node.attrib) == 1
+                and "rank" in node.attrib
+            ):
+                dim_index_value.append(node.attrib["rank"])
```

Comment:
Here we are handling only the `dim` nxdl node. Rank is an attribute of `dimension` node, which already handled bellow, e.g. 
```python
                for attr, value in node.attrib.items():
                    if attr in possible_dimension_attrs and not isinstance(value, dict):
                        indent = (depth + 1) * DEPTH_SIZE
                        file_out.write(f"{indent}{attr}: {value}\n")
```

Reply:
done